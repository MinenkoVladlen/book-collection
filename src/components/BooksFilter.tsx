import { Box, FormControlLabel, Switch, Typography } from '@mui/material';
import React from 'react';
import booksStore from '../services/books/Books.store.ts';

const BooksFilter: React.FC = () => {

    const handleChange = (e) => {
        booksStore.toggleShowPrivate()
    }

    const {showPrivate} = booksStore;

    return (
        <Box className='header'>
            <FormControlLabel 
                control={<Switch checked={showPrivate} onChange={handleChange} />} 
                label="Show only private" 
            />
            <Typography variant='body1'>Total Books: {booksStore.books.length}</Typography>
        </Box>
    )
}

export default BooksFilter;