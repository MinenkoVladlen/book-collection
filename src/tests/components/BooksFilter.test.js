import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import BooksFilter from '../../components/BooksFilter';
import booksStore from '../../services/books/Books.store';

jest.mock('../../services/books/Books.store');

describe('BooksFilter', () => {
  beforeEach(() => {
    booksStore.books = [];
  });

  it('should render correctly', () => {
    const { asFragment } = render(<BooksFilter />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should toggle showPrivate when button is clicked', async () => {
    const { getByText } = render(<BooksFilter />);
    fireEvent.click(getByText(/Show only private/i));
    expect(booksStore.toggleShowPrivate).toHaveBeenCalled();
  });
});