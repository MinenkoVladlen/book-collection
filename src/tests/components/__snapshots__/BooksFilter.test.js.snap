// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksFilter should render correctly 1`] = `
<DocumentFragment>
  <div
    class="header MuiBox-root css-0"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        Show only private
      </span>
    </label>
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      Total Books: 0
    </p>
  </div>
</DocumentFragment>
`;
